### PR TITLE
Accomodating an extra widget with rewards on NTP

### PR DESCRIPTION
**Pull request by ryanml**
Fixes: https://github.com/brave/brave-browser/issues/8647

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
1. Confirm via the new tab storybook that widget stack is correctly visible
2. As a new widget has not been added yet, there should be no changes to the NTP when building the browser. (Confirm it so)
3. To confirm the tab like functionality, set this prop to `true` for the `<NewTabPage>` component in use, and build to test: https://github.com/brave/brave-core/pull/4911/files#diff-4a955c5803d3401941f275f21d0d3ac1R29
5. Clicking the title tab of the widget behind Rewards should put it in to the foreground, vice versa for rewards.
6. Widget order should persist.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/brave_new_tab_ui/components/default/exchange/style.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+  isInTab?: boolean
+}
+
+export const StyledTitleTab = styled<StyleProps, 'div'>('div')`
+  color: #fff;
+  cursor: pointer;
+  padding: 10px ${p => p.isInTab ? 13 : 25}px;
+  margin-bottom: -3px;
+  border-radius: 6px 6px 0 0;
+  background-image: linear-gradient(270deg, rgba(33,37,41,0.16) 0%, #212529 100%);
```
Comment (petemill) [inline components/brave_new_tab_ui/components/default/exchange/style.ts]: I think backdrop filter will get a nicer glass-like effect that also the design seems to call for

File: components/brave_new_tab_ui/components/default/exchange/assets/exchange-logo.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+
```
Comment (cezaraugusto) [inline components/brave_new_tab_ui/components/default/exchange/assets/exchange-logo.tsx]: nit: extra line

File: components/brave_new_tab_ui/components/default/exchange/index.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+      </StyledTitleTab>
+    )
+  }
+
+  renderBuyView () {
+    return (
+      <>
+        <Content>
+          <Copy>
+            {'Buy Crypto'}
```
Comment (cezaraugusto) [inline components/brave_new_tab_ui/components/default/exchange/index.tsx]: better if locale-ready

File: components/brave_new_tab_ui/components/default/exchange/index.tsx [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+                {'BTC'}
+              </AssetDropdownLabel>
+              <CaratDropdown>
+                <CaratDownIcon />
+              </CaratDropdown>
+            </AssetDropdown>
+          </BuyPromptWrapper>
+          <ActionsWrapper>
+            <ConnectButton>
+              {'Buy BTC'}
```
Comment (cezaraugusto) [inline components/brave_new_tab_ui/components/default/exchange/index.tsx]: ditto

File: components/brave_new_tab_ui/components/default/exchange/style.ts [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+`
+export const WidgetWrapper = styled<{}, 'div'>('div')`
+  color: white;
+  padding: 10px 15px;
+  border-radius: 6px;
+  position: relative;
+  font-family: Muli, sans-serif;
```
Comment (cezaraugusto) [inline components/brave_new_tab_ui/components/default/exchange/style.ts]: let's use props from theme `{p => p.theme.fontFamily.body}`. ref https://github.com/brave/brave-ui/blob/master/src/theme/brave-default.ts#L60

File: components/brave_new_tab_ui/components/default/exchange/style.ts [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+  border: 1px solid rgb(70, 70, 70);
+  margin-bottom: 10px;
+`
+export const FiatDropdown = styled<{}, 'div'>('div')`
+  float: right;
+  width: 25%;
+  padding: 7px 3px 0px 7px;
+`
+
+export const FiatDropdownLabel = styled<{}, 'span'>('span')`
```
Comment (cezaraugusto) [inline components/brave_new_tab_ui/components/default/exchange/style.ts]: I guess this could be removed?

Comment (ryanml) [reply components/brave_new_tab_ui/components/default/exchange/style.ts]: removed

File: components/brave_new_tab_ui/components/default/exchange/style.ts (hunk 5 above)
Comment (ryanml) [reply components/brave_new_tab_ui/components/default/exchange/style.ts]: done

File: components/brave_new_tab_ui/components/default/exchange/assets/exchange-logo.tsx (hunk 2 above)
Comment (ryanml) [reply components/brave_new_tab_ui/components/default/exchange/assets/exchange-logo.tsx]: removed

File: components/brave_new_tab_ui/containers/newTab/index.tsx [hunk 7]
```diff
@@ -230,7 +234,69 @@ class NewTabPage extends React.Component<Props, State> {
...
+  }
+
+  // Widget toggling/preventFocus tied to rewards for now
+  // This will be updated when we integrate another widget
+  renderCryptoContent () {
+    const { newTabData } = this.props
+    const { textDirection, showRewards } = newTabData
+
+    return (
+      <Page.GridItemRewards>
```
[the remark this answers is not in the document]
Comment (ryanml) [reply components/brave_new_tab_ui/containers/newTab/index.tsx]: agreed, renamed

File: components/brave_new_tab_ui/containers/newTab/index.tsx [hunk 8]
```diff
@@ -230,7 +234,69 @@ class NewTabPage extends React.Component<Props, State> {
...
-  renderRewardsContent () {
+  toggleStackWidget = (widgetId: NewTab.StackWidget) => {
+    this.props.actions.setCurrentStackWidget(widgetId)
+  }
+
+  getCryptoContent () {
+    const { newTabData, useTestingWidget } = this.props
+    const { currentStackWidget } = newTabData
+
+    if (!useTestingWidget) {
```
[the remark this answers is not in the document]
Comment (cezaraugusto) [reply components/brave_new_tab_ui/containers/newTab/index.tsx]: +1 best if we could add it to `storybookState.ts` file and add a boolean knob there. Similar to https://github.com/brave/brave-core/blob/master/components/brave_new_tab_ui/stories/default/data/storybookState.ts#L51

File: components/brave_new_tab_ui/stories/default/data/storybookState.ts [hunk 9]
```diff
@@ -46,6 +46,7 @@ function shouldShowBrandedWallpaperData (shouldShow: boolean) {
 
 export const getNewTabData = (state: NewTab.State) => ({
   ...state,
+
```
Comment (ryanml) [inline components/brave_new_tab_ui/stories/default/data/storybookState.ts]: nit: extra line

Comment (cezaraugusto) [reply components/brave_new_tab_ui/stories/default/data/storybookState.ts]: Oops, fixed

File: components/brave_new_tab_ui/components/default/rewards/index.tsx [hunk 10]
```diff
@@ -33,6 +32,7 @@ import {
   TurnOnAdsButton,
   UnsupportedMessage
 } from './style'
+import { StyledTitleTab } from '../exchange/style'
```
[the remark this answers is not in the document]
Comment (ryanml) [reply components/brave_new_tab_ui/components/default/rewards/index.tsx]: done

File: components/brave_new_tab_ui/components/default/exchange/style.ts (hunk 1 above)
Comment (ryanml) [reply components/brave_new_tab_ui/components/default/exchange/style.ts]: added

File: components/brave_new_tab_ui/containers/newTab/index.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (ryanml) [reply components/brave_new_tab_ui/containers/newTab/index.tsx]: the test widget has been removed

File: components/brave_new_tab_ui/components/default/exchange/index.tsx (hunk 3 above)
Comment (ryanml) [reply components/brave_new_tab_ui/components/default/exchange/index.tsx]: test widget removed

File: components/brave_new_tab_ui/components/default/exchange/index.tsx (hunk 4 above)
Comment (ryanml) [reply components/brave_new_tab_ui/components/default/exchange/index.tsx]: test widget removed

**Comment by bsclifton** [comment]
CI looks good! There were some warnings that are unrelated (Lint was already fixed in master, same w/ Speedreader related browser test failure). Going to go ahead and merge! 😄

**Pull request merged**